src/Data/Record.purs

```diff
@@ -109,3 +113,31 @@ delete
   -> Record r2
   -> Record r1
 delete l r = runFn2 unsafeDeleteFn (reflectSymbol l) r
+
+-- | Check two records of the same type for equality.
+equal
+  :: forall r rs
+   . RowToList r rs
+  => EqualFields rs r
+  => Record r
+  -> Record r
+  -> Boolean
+equal a b = equalFields (RLProxy :: RLProxy rs) a b
+
+class EqualFields (rs :: RowList) (row :: # Type) | rs -> row where
+  equalFields :: RLProxy rs -> Record row -> Record row -> Boolean
+
+instance equalFieldsCons
+  ::
+  ( IsSymbol name
+  , Eq ty
+  , RowCons name ty tailRow row
+  , EqualFields tail row
+  ) => EqualFields (Cons name ty tail) row where
+  equalFields _ a b = get' a == get' b && rest
```

Comment:
hey @justinwoo! I think it might be more efficient to use an `if` expression to make this short-circuit:
```purescript
  equalFields _ a b =
    if get' a == get' b
      then equalRest a b -- evaluate this fully only inside the true branch
      else false
    where
      get' = get (SProxy :: SProxy name)
      equalRest = equalFields (RLProxy :: RLProxy tail)
```
(I _think_ the strictness means that the rest of the record would be compared before being passed to `&&`)

Reply:
`get' a == get' b && equalRest a b` would also short circuit :)

Reply:
Oh, okay! Sometimes I forget we have compiler magic – thanks! :)

Reply:
updated! thanks